src/Sentry.Unity.Android/SentryJava.cs

```diff
@@ -40,6 +42,55 @@ internal class SentryJava : ISentryJava
 {
     private static AndroidJavaObject GetSentryJava() => new AndroidJavaClass("io.sentry.Sentry");
 
+    public void Init(IJniExecutor jniExecutor, SentryUnityOptions options)
+    {
+        jniExecutor.Run(() =>
+        {
+            using var sentry = new AndroidJavaClass("io.sentry.android.core.SentryAndroid");
+            using var context = new AndroidJavaClass("com.unity3d.player.UnityPlayer")
+                .GetStatic<AndroidJavaObject>("currentActivity");
+
+            sentry.CallStatic("init", context, new OptionsConfiguration(androidOptions =>
+            {
+                androidOptions.Call("setDsn", options.Dsn);
+                androidOptions.Call("setDebug", options.Debug);
+                androidOptions.Call("setRelease", options.Release);
+                androidOptions.Call("setEnvironment", options.Environment);
+            }, options.DiagnosticLogger));
+        });
+    }
+
+    // Update the callback class to match the Java interface name
+    internal class OptionsConfiguration : AndroidJavaProxy
+    {
+        private readonly Action<AndroidJavaObject> _callback;
+        private readonly IDiagnosticLogger? _logger;
+
+        public OptionsConfiguration(Action<AndroidJavaObject> callback, IDiagnosticLogger? logger)
+            : base("io.sentry.Sentry$OptionsConfiguration")
```

Comment:
Any concerns that proguard can mess with this? I imagine not, because we have proguard rules. Assuming those get passed in to the final build.
I imagine this is already a concern to all the other calls we do into Java

Reply:
I don't think so. We're using the very same mechanism for the scope sync so far tho.